avm/ptn/dev-ops/cicd-agents-and-runners/main.bicep

```diff
@@ -0,0 +1,1030 @@
+metadata name = 'CI CD Agents and Runners'
+metadata description = 'This module deploys self-hosted agents and runners for Azure DevOps and GitHub on Azure Container Instances and/or Azure Container Apps.'
+metadata owner = 'Azure/module-maintainers'
+
+// ================ //
+// Parameters       //
+// ================ //
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. Naming prefix to be used with naming the deployed resources.')
+param namingPrefix string
+
+@description('Required. The compute target for the private runners.')
+param computeTypes computeTypesType
+
+@description('Required. The self-hosted runner configuration. This can be either GitHub or Azure DevOps.')
+param selfHostedConfig selfHostedRunnerType
+
+@description('Required. The networking configuration.')
+param networkingConfiguration networkType
+
+@description('Optional. Whether to use private or public networking for the Azure Container Registry.')
+param privateNetworking bool = true
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+// ================ //
+// Variables        //
+// ================ //
+
+var imagePaths = [
+  {
+    platform: 'azuredevops-container-app'
+    imagePath: 'azure-devops-agent-aca'
+  }
+  {
+    platform: 'github-container-app'
+    imagePath: 'github-runner-aca'
+  }
+  {
+    platform: 'azuredevops-container-instance'
+    imagePath: 'azure-devops-agent-aci'
+  }
+  {
+    platform: 'github-container-instance'
+    imagePath: 'github-runner-aci'
+  }
+]
+
+// To be updated
+var githubURL = 'https://github.com/sebassem/bicep-registry-modules#avm-ptn-lz-private-runners:avm/ptn/dev-ops/cicd-agents-and-runners/container-images'
```

Comment:
This sounds ominous 😄  Also, should it be a parameter?

Reply:
I missed that one 😄. This value will not change, it should always point to upstream to get the images. I will update to point to upstream

Reply:
Resolved